### PR TITLE
Nerdrew/truncate old floor dates

### PR DESCRIPTION
**Pull request by nerdrew**
@mitchel456 @yellow-beard

### CODE REVIEW

File: rrule.gemspec [hunk 1]
```diff
@@ -23,5 +23,4 @@ Gem::Specification.new do |s|
 
   s.required_ruby_version = '>= 2.3.0'
   s.add_runtime_dependency 'activesupport', '>= 4.1'
-  s.add_development_dependency 'rspec', '~> 3.4'
```
Comment (mitchel456) [inline rrule.gemspec]: What's the reason for putting this in the gemfile? You shouldn't have to install rspec just to use the gem, right?

File: Gemfile [hunk 2]
```diff
@@ -1,5 +1,11 @@
 source 'https://rubygems.org'
 
+gem 'pry'
```
Comment (mitchel456) [inline Gemfile]: Not sure we should make `pry` a dependency - you don't need it to run specs, and it'd be nice to keep dependencies to a minimum. Not hard to just install pry yourself if you want to use it when debugging.

File: rrule.gemspec (hunk 1 above)
Comment (nerdrew) [reply rrule.gemspec]: Gemfile is ignored for users of the gem. It only matters for developing the gem. `add_development_dependency` vs putting in the Gemfile are functionally equivalent.

File: Gemfile (hunk 2 above)
Comment (nerdrew) [reply Gemfile]: Eh. I like having it in all my gems when debugging them, and it adds no runtime dependency. If you feel strongly, I'll remove it.

File: rrule.gemspec (hunk 1 above)
Comment (mitchel456) [reply rrule.gemspec]: Got it - sounds good.

File: Gemfile (hunk 2 above)
Comment (mitchel456) [reply Gemfile]: Nah, I don't feel super strongly, just a (possibly mostly aesthetic) objection that it's not an _actual_ dependency, just a nice to have that people could add themselves if they wanted to, but NABD.

**Pull request merged**